evap/contributor/views.py

```diff
@@ -28,7 +28,7 @@ def index(request):
     all_courses.sort(key=lambda course: list(STATES_ORDERED.keys()).index(course.state))
 
     for course in all_courses:
-        course.distribution = calculate_average_distribution(course) if course.can_user_see_grades(user) else None
+        course.distribution = calculate_average_distribution(course) if course.has_enough_voters_to_publish_grade(user) else None
```

Comment:
an assert inside calculate_average_distribution that `course.has_enough_voters_to_publish_grade` would be nice.

also, there is no `user` parameter to `has_enough_voters_to_publish_grade`